catroid/src/org/catrobat/catroid/physics/PhysicsLook.java

```diff
@@ -128,10 +133,26 @@ public void setRotation(float degrees) {
 
 	@Override
 	public void setScale(float scaleX, float scaleY) {
+		Vector2 oldScales = new Vector2(getScaleX(), getScaleY());
+		if (scaleX < 0.0f || scaleY < 0.0f) {
+			scaleX = 0.0f;
+			scaleY = 0.0f;
+		}
+
+		int scaleXComp = Math.round(scaleX * SCALEFACTOR_ACCURACY);
+		int scaleYComp = Math.round(scaleY * SCALEFACTOR_ACCURACY);
+		if (scaleXComp == Math.round(oldScales.x * SCALEFACTOR_ACCURACY) && scaleYComp == Math.round(oldScales.y * SCALEFACTOR_ACCURACY)) {
+			return;
+		}
+
+		scaleX = scaleXComp / SCALEFACTOR_ACCURACY;
+		scaleY = scaleYComp / SCALEFACTOR_ACCURACY;
 		super.setScale(scaleX, scaleY);
```

Comment:
You're right, of course. I'll remove this but am undecided as to whether a check should be performed which would prevent "re-setting" the same scale (as in line 144). This would require a float comparison delta which would then also disallow minuscule changes in scale (which could of course add up to larger numbers). What's your recommendation?